# ava/no-invalid-modifier-chain

ğŸ“ Disallow invalid modifier chains.

ğŸ’¼ This rule is enabled in the âœ… `recommended` [config](https://github.com/avajs/eslint-plugin-ava#recommended-config).

ğŸ”§ğŸ’¡ This rule is automatically fixable by the [`--fix` CLI option](https://eslint.org/docs/latest/user-guide/command-line-interface#--fix) and manually fixable by [editor suggestions](https://eslint.org/docs/latest/use/core-concepts#rule-suggestions).

<!-- end auto-generated rule header -->

Translations: [FranÃ§ais](https://github.com/avajs/ava-docs/blob/main/fr_FR/related/eslint-plugin-ava/docs/rules/no-invalid-modifier-chain.md)

AVA only allows specific [test modifier](https://github.com/avajs/ava/blob/main/docs/01-writing-tests.md) chains. Using modifiers in the wrong order, combining incompatible modifiers, or using modifiers that don't apply to a given test type will cause runtime errors.

## Examples

```js
import test from 'ava';

// Wrong order
test.only.serial(t => {});  // âŒ
test.serial.only(t => {});  // âœ…

test.failing.serial(t => {}); // âŒ
test.serial.failing(t => {}); // âœ…

// Invalid combinations
test.only.skip(t => {});  // âŒ

// Invalid modifiers on hooks
test.before.failing(t => {}); // âŒ
test.before.only(t => {});    // âŒ
test.before(t => {});         // âœ…

// Invalid todo chains
test.todo.failing('title'); // âŒ
test.todo('title');         // âœ…

// Invalid always usage
test.before.always(t => {});    // âŒ `.always` only works with `after`/`afterEach`
test.after.always(t => {});     // âœ…
test.afterEach.always(t => {}); // âœ…

// Unknown modifiers
test.foo(t => {}); // âŒ
test.cb(t => {});  // âŒ

// Duplicates
test.serial.serial(t => {}); // âŒ
test.serial(t => {});        // âœ…
```
